models/community.js

```diff
@@ -0,0 +1,14 @@
+const mongoose = require('mongoose');
+
+const CommunitySchema = new mongoose.Schema({
+  name: { type: String, unique: true, maxlength: 20, trim: true, required: true },
+  creator_id: { type: String },
+  group: { type: String }, // group Id or public name
+  channel1: { type: String }, // channel 1 Id or public name, usually buy, if only one channel is used then buy/sell
```

Comment:
`order_channels: [OrderChannelSchema]` with custom validation for array length (0 < arr.length < 2).

```js
const OrderChannelSchema = new mongoose.Schema({
    name: String,
    type: {
        type: String,
        enum: { values: ['buy', 'sell', 'mixed'], message: '{VALUE} is not supported' }
     }
}
```

Reply:
I like this approach more, it is more concise and clear